Queueomatic/Server/Services/RoomDeletionService/RoomDeletionService.cs

```diff
@@ -0,0 +1,31 @@
+﻿using Queueomatic.DataAccess.UnitOfWork;
+
+namespace Queueomatic.Server.Services.RoomDeletionService;
+
+public sealed class RoomDeletionService : IRoomDeletionService
+{
+    private readonly IUnitOfWork _unitOfWork;
+    private readonly ILogger<RoomDeletionService> _logger;
+
+    public RoomDeletionService(IUnitOfWork unitOfWork, ILogger<RoomDeletionService> logger)
+    {
+        _unitOfWork = unitOfWork;
+        _logger = logger;
+    }
+
+    public async Task DeleteExpiredRoomsAsync()
+    {
+        var roomsToBeDeleted = _unitOfWork.RoomRepository
+            .GetAllAsync()
```

Comment:
Do an overload instead of making changes to it after. It is much more efficient to directly filter on the dbcontext.